extension/intents/pocket/pocket.js

```diff
@@ -0,0 +1,27 @@
+import * as pageMetadata from "../../background/pageMetadata.js";
+import * as intentRunner from "../../background/intentRunner.js";
+import * as content from "../../background/content.js";
+
+intentRunner.registerIntent({
+  name: "pocket.open",
+  async run(context) {
+    const url = "https://app.getpocket.com/";
+    await context.createTab({ url });
+  },
+});
+
+intentRunner.registerIntent({
+  name: "pocket.save",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const metadata = await pageMetadata.getMetadata(activeTab.id);
+    await content.lazyInject(activeTab.id, [
+      "/background/pageMetadata-contentScript.js",
+      "/intents/pocket/pocketContentScript.js",
+    ]);
+    await browser.tabs.sendMessage(activeTab.id, {
+      type: "save",
+      url: metadata.url,
+    });
```

Comment:
This stuff doesn't actually have to happen inside the page itself, it can happen right here. That is, you can try to get the save URL right here.

Reply:
It might be even better to open the save URL in a hidden tab, and then close it if it goes OK, and focus it if not (e.g., when you aren't logged in).